joblib/test/test_hashing.py

```diff
@@ -70,8 +72,8 @@ def f(self, x):
 
 class KlassWithCachedMethod(object):
 
-    def __init__(self):
-        mem = Memory(cachedir=test_memory_env['dir'])
+    def __init__(self, cachedir):
+        mem = Memory(cachedir=cachedir)
```

Comment:
I changed the parameter name from tmpdir_path to cachedir. Less chance for confusion this way.